lib/wickedbase.pm

```diff
@@ -489,17 +489,35 @@ sub setup_team {
     $self->get_from_data('wicked/ifcfg/ifcfg-eth0-hotplug',     $cfg_ifc0);
     $self->get_from_data('wicked/ifcfg/ifcfg-eth0-hotplug',     $cfg_ifc1);
     $self->get_from_data('wicked/teaming/ifcfg-team0-' . $mode, $cfg_team0);
-    file_content_replace($cfg_team0, ipaddr4 => $self->get_ip(type => 'host', netmask => 1), ipaddr6 => $self->get_ip(type => 'host6', netmask => 1), iface0 => $iface0, iface1 => $iface1);
+    my $ipaddr4  = $self->get_ip(type => 'host',  netmask       => 1);
+    my $ipaddr6  = $self->get_ip(type => 'host6', netmask       => 1);
+    my $ping_ip4 = $self->get_ip(type => 'host',  is_wicked_ref => 1);
+    my $ping_ip6 = $self->get_ip(type => 'host6', is_wicked_ref => 1);
+    file_content_replace($cfg_team0, ipaddr4 => $ipaddr4, ipaddr6 => $ipaddr6, iface0 => $iface0, iface1 => $iface1, ping_ip4 => $ping_ip4, ping_ip6 => $ping_ip6);
 
     $self->wicked_command('ifup', 'all');
 }
 
+sub get_team_active_link {
+    my ($self, $team_link) = @_;
+    my $out = script_output("teamdctl $team_link state view");
+    record_info('teamdctl', $out);
+    my @lines = split(/\n/, $out);
```

Comment:
```suggestion
    return $1 if ( $out =~ m/active port: (\w+)/);
    return;
```

Reply:
ok, looks much cleaner. Thanks :)

Reply:
is second return is really necessary ? I think we will getting undef anyway , right ?

Reply:
These are dark perl internals... :)
I would think we are getting `0`, as we retrive the outcome of the last statement which is =~ m//. But don't know without reading the doc again. So adding that return makes it clear to me.